doc/bfd/BFD HW Offload for BGP session HLD.md

```diff
@@ -0,0 +1,608 @@
+
+One way to get destination mac address is to read neighbor table, for example, using cmd ip -6 neighbor.
+```
+sonic@sonic:/var/tmp$ ip -6 neighbor get fe80::7a6b:17ff:fe5a:7000 dev Ethernet0
+fe80::7a6b:17ff:fe5a:7000 dev Ethernet0 lladdr 78:6b:17:5a:70:00 router REACHABLE
+sonic@sonic:/var/tmp$ 
+```
+
+To make sure the the destination mac address is available before bgp creating bfd session, do not use this bgp configuration: "neighbor xxx disable-connected-check", let bgp check connection first before it try to create bfd session. so peer system mac address will be available if bfdsyncd issue a PING before try to read the neighbor table.
+
+example for bfd session with link local address:
+```
+bgp configuration:
+sonic(config-router)#         neighbor FOO peer-group
+sonic(config-router)#         neighbor FOO remote-as external
+sonic(config-router)#         neighbor FOO ebgp-multihop 1
+sonic(config-router)#         neighbor FOO bfd
+sonic(config-router)#         neighbor Ethernet0 interface peer-group FOO
+
+sonic@sonic:/var/tmp$ show bfd sum
+Total number of BFD sessions: 1
+Peer Addr                  Interface    Vrf      State    Type          Local Addr                   TX Interval    RX Interval    Multiplier  Multihop      Local Discriminator
+-------------------------  -----------  -------  -------  ------------  -------------------------  -------------  -------------  ------------  ----------  ---------------------
+fe80::7a6b:17ff:fe5a:7000  Ethernet0    default  Up       async_active  fe80::7a3e:86ff:fe14:8400            300            300             3  false                           2
+sonic@sonic:/var/tmp$ vtysh -c "show bfd peer"
+BFD Peers:
+	peer fe80::7a6b:17ff:fe5a:7000 local-address fe80::7a3e:86ff:fe14:8400 vrf default interface Ethernet0
+		ID: 2602968139
+		Remote ID: 1
+		Active mode
+		Status: up
+		Uptime: 26 second(s)
+		Diagnostics: ok
+		Remote diagnostics: ok
+		Peer Type: dynamic
+		RTT min/avg/max: 0/0/0 usec
+		Local timers:
+			Detect-multiplier: 3
+			Receive interval: 300ms
+			Transmission interval: 300ms
+			Echo receive interval: 50ms
+			Echo transmission interval: disabled
+		Remote timers:
+			Detect-multiplier: 3
+			Receive interval: 300ms
+			Transmission interval: 300ms
+			Echo receive interval: disabled
+
+```
+
+## 2.8 Control plane BFD
+
+A control plane BFD approach is to use FRR SW BFD. If use HW offload BFD for FRR, it is expected that all FRR BFD sessions are offloaded to HW, with a global flag to switch when start bfdd. 
+
+## 2.9 CLI
+
+no new CLI introduced for this feature
+
+# 3 Configuration
+To support hardware bfd offload for bgp,need to run two daemons inside bgp container:
+ bfdd --dplaneaddr ipv4c:127.0.0.1
+ bfdsyncd
+But the logic to launch is outside of the scope of this document. 
+Here is an example to configure it in config_db:
+
+/etc/sonic/config_db.json
+```
```

Comment:
Can this be enabled through 'config feature state' CLI? If so, could you add yang model changes as well.